src/backend/tasks_io/datafeeds/datafeed_fms_api.py

```diff
@@ -321,22 +321,30 @@ def _merge_match_schedule_and_results(self, schedule: Dict, matches: Dict) -> Di
 
     @classmethod
     def _merge_match(cls, scheduled: Dict, result: Dict) -> Dict:
+        # 2024, Week 4: As part of attempting to restore sync,
+        # schedules sync with teams [1, 2, 3] in to-be-played playoff matches
+        # In a case where the only teams for a match are those three, overwrite
+        # the team numbers in each station to None
+        schedule_team_numbers = [t["teamNumber"] for t in scheduled.get("teams")]
+        if set(schedule_team_numbers) == {1, 2, 3}:
+            scheduled["teams"] = [{**t, "teamNumber": None} for t in scheduled.get("teams")]
+
         for field, value in result.items():
             if field == "teams":
-                for i, team in enumerate(value):
-                    schedule_team = next(
+                for team in value:
+                    schedule_idx, schedule_team = next(
                         filter(
-                            lambda t: t["teamNumber"] == team["teamNumber"],
-                            scheduled["teams"],
+                            lambda t: t[1]["teamNumber"] == team["teamNumber"],
+                            enumerate(scheduled["teams"]),
                         ),
-                        None,
+                        (None, None),
                     )
                     if schedule_team is None:
                         # 2024, Week 3: Upstream FMS sync issues leading to schedules returned with no teams
                         # Some match results have been sync'd, so patch around this where we can
                         scheduled["teams"].append(team)
                     else:
-                        scheduled["teams"][i].update(team)
```

Comment:
For cases where an alliance uses a lineup card and uses a non-default station assignment, we want to make sure we call `update` on the corresponding team number, which can update the station assignment

(for this reason, we extract the index in the `scheduled` list above)